tests/foreman/api/test_provisioningtemplate.py

```diff
@@ -406,7 +404,65 @@ def test_positive_template_check_vlan_parameter(
                 },
             ],
         ).create()
-        provision_template = host.read_template(data={'template_kind': 'provision'})
-        assert f'interfacename=vlan{tag}' in str(provision_template)
-        ipxe_template = host.read_template(data={'template_kind': 'iPXE'})
-        assert f'vlan=vlan{tag}:{identifier}' in str(ipxe_template)
+        provision_template = host.read_template(data={'template_kind': 'provision'})['template']
+        assert f'interfacename=vlan{tag}' in provision_template
+        ipxe_template = host.read_template(data={'template_kind': 'iPXE'})['template']
+        assert f'vlan=vlan{tag}:{identifier}' in ipxe_template
+
+    @pytest.mark.parametrize('module_sync_kickstart_content', [7, 8, 9], indirect=True)
+    @pytest.mark.parametrize('pxe_loader', ['uefi'], indirect=True)
+    @pytest.mark.parametrize('boot_mode', ['Static', 'DHCP'])
+    def test_positive_template_subnet_with_boot_mode(
+        self,
+        module_sync_kickstart_content,
+        module_target_sat,
+        module_sca_manifest_org,
+        module_location,
+        default_architecture,
+        default_partitiontable,
+        pxe_loader,
+        boot_mode,
+    ):
+        """Check whether boot mode paremeter in subnet respected when PXELoader UEFI is used,
+            and properly rendered in the provisioning templates
+
+        :id: 2decc787-59b0-41e6-96be-5dd9371c8966
+
+        :expectedresults: templates should get render and contains boot mode used as set in subnet
+
+        :BZ: 2168967, 1955861, 1784012
+
+        :customerscenario: true
+        """
+        subnet = module_target_sat.api.Subnet(
+            name=gen_string('alpha'),
+            organization=[module_sca_manifest_org],
+            location=[module_location],
+            network='192.168.0.1',
+            mask='255.255.255.240',
+            boot_mode=boot_mode,
+        ).create()
+        host = module_target_sat.api.Host(
+            name=gen_string('alpha'),
+            organization=module_sca_manifest_org,
+            location=module_location,
+            subnet=subnet,
+            pxe_loader=pxe_loader.pxe_loader,
+            root_pass=settings.provisioning.host_root_password,
+            ptable=default_partitiontable,
+            architecture=default_architecture,
+            operatingsystem=module_sync_kickstart_content.os,
+        ).create()
+        # Verify provision templates for boot_mode in subnet, and check provision logs exists
+        rendered = host.read_template(data={'template_kind': 'provision'})['template']
+        assert f'--bootproto {boot_mode.lower()}' in rendered
+        assert '/root/install.post.log' in rendered
+        assert '/mnt/sysimage/root/install.post.log' not in rendered
+
+        # Verify PXE templates for boot_mode in subnet
+        pxe_templates = ['PXEGrub', 'PXEGrub2', 'PXELinux', 'iPXE']
+        provision_url = f'http://{module_target_sat.hostname}/unattended/provision'
```

Comment:
I don't see BZ 2168967 (escaping of `&`) actually tested here, can you explain?

Reply:
Yes sure, before this BZ, PXEGrub2 template wasn't rendering for UEFI when boot_mode==Static and provisioning fails, but it works now so we're just testing the template render here, and I included all other PXE templates to have better coverage around this scenario.

Reply:
In the downloaded template, should the `&` be escaped? Or is it in some other place? If it should be escaped, can you test it, as an addition to testing that the template just renders?

Reply:
ah, I got you know, so you're mentioning about `options.push("ks=#{foreman_url('provision', static6: '1').gsub("&", "\\\\&")}")` from original PR, so this is adding `'?static=1'` in the end of provision_url, and we're testing the very same URL is rendered correctly in the test at last `assert f'ks={ks_param}' in rendered`

@lhellebr If you still think if I misunderstood this, so please feel free to open a separate PR to add this additional assertion, I'd very much appreciate it, Thank you!